### PR TITLE
Acceleration/Jerk override on layer change

### PR DESCRIPTION
**Pull request by kyjak**
When using Jerk/Acceleration settings in Cura, it can have a destroying effect on some printers.
Confirmed case is the Prusa MK3 printer (confirmed by Prusa technical support), which cannot handle these, because they happen too often.
As a workaround, this change request adds possibility for the plugin to set acceleration/jerk at the start of a layer, for example the initial layer, which sometimes needs less acceleration to better bed adhesion.

### CODE REVIEW

**Comment by Ghostkeeper** [comment]
As far as I can tell from the implementation, this will only change the acceleration and jerk if Acceleration Control and Jerk Control respectively are disabled in the settings, right?

**Comment by kyjak** [comment]
Dont know how to check for the main settings page if it is enabled/disabled. It should be up to the user knowledge if to use this override at layer or not. If someone uses Curas jerk/acc settings, this setting, if enabled and used, will render useless and will be overruled byt curas jerk/acc realtime settings.

**Comment by kyjak** [comment]
Maybe I should write a better "readme" / hints?

**Comment by Ghostkeeper** [comment]
Yeah there is really no place to put that "readme" currently. But we can at least mention this in the setting descriptions.

**Comment by kyjak** [comment]
I have upgraded the usage help strings, should be fine now.

File: plugins/PostProcessingPlugin/scripts/ChangeAtZ.py [hunk 1]
```diff
@@ -279,6 +281,36 @@ def getSettingDataString(self):
...
                     "enabled": "j1_Change_fanSpeed"
+                },
+                "k1_Change_acceleration":
+                {
+                    "label": "Change acceleration",
+                    "description": "Select if you want to change max. acceleration. The setting will be overriden when you also use Cura profile acc. settings. Works only with layer number, not height.",
+                    "type": "bool",
+                    "default_value": false
+                },
+                "k2_acceleration":
```
Comment (ninovanhooff) [inline plugins/PostProcessingPlugin/scripts/ChangeAtZ.py]: Looking at other parameters, I expected this to be named k3_ and the next one k4_

**Comment by Ghostkeeper** [comment]
This needs significant changes now that the Change at Z script was largely rewritten for 5.2.

But I also don't think that we should accept this change if it only works when providing a layer number, not when providing a Z height. Most people don't read tooltips, so they will expect it to work fine just like all of the other settings. There is really no good reason why this should only work when pausing at a certain Z coordinate.

**Comment by kyjak** [comment]
Ok, will merge current branch and try to adopt the script, also add layer height not only index. If you want to close this PR in the meantime is fine, can create new one when done.

**Pull request closed without merging**